### PR TITLE
Make full train compatible with updated library

### PR DESCRIPTION
**Pull request by Maxusmusti**
Makes the full train script slightly more resistant to upcoming instructlab-training library changes, and ensures compatibility with both the current version and post-https://github.com/instructlab/training/pull/257

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the
  [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [ ] [Changelog](https://github.com/instructlab/instructlab/blob/main/CHANGELOG.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Functional tests have been added, if necessary.
- [ ] E2E Workflow tests have been added, if necessary.

### CODE REVIEW

File: src/instructlab/model/full_train.py [hunk 1]
```diff
@@ -144,7 +144,7 @@ def train(train_args, device):
         )
         model.config.eos_token_id = tokenizer.eos_token_id
 
-    model = utils.convert_loss_to_reduce_sum(model, is_granite=False)
```
Comment (cdoern) [inline src/instructlab/model/full_train.py:147]: Does this still work?

**Review by cdoern — COMMENTED** [review]
one comment about removing the is_granite arg from one of the util function calls

File: src/instructlab/model/full_train.py (hunk 1 above)
Comment (Maxusmusti) [reply src/instructlab/model/full_train.py:147]: @cdoern Yeah, this is the old function header in utils currently:
```
def convert_loss_to_reduce_sum(model, is_granite=False):
```

and this is the new one:
```
def convert_loss_to_reduce_sum(model, use_dolomite=False):
```

so in both cases the default suffices

Comment (Maxusmusti) [reply src/instructlab/model/full_train.py:147]: This way we also avoid `is_granite` breaking with the upcoming release, and `use_dolomite` breaking with the current release

**Review by cdoern — APPROVED** [review]

**Pull request merged**